### ISSUE TITLE
print contract button breaks on refinanced contract

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1636]
_Imported from trac ticket 1636.
Created by **sal**
Opened in trac: May 7, 2012, 4:36:12 PM
Last modified in trac: May 31, 2012, 12:25:01 PM_
***
_by sal:_
Print contract button breaks on a refinanced contract. Once you refresh your browser you can then print a contract but it always fails if you try and print a refinanced contract

**Comment by kbower** [comment]
_**rneville** trac comment on May 8, 2012, 9:23:10 AM:_

The problem is that the installment contract report _requires_ as its sole bit of input the order id to which it is attached. When a contract is refinanced, the SAL transaction of the original contract is not updated to the current contractid, thus:
```python
    @memoized_property
    def currentorders(self):
        #return DBSession.query(Order)\
            #.filter(Order.transactions.any(and_(ArTranBase.transactioncodeid==ArTran.SALE,ArTranBase.contractid==self.contractid)))\
            #.all()
        return DBSession.query(Order).enable_eagerloads(False)\
            .filter(and_(ArTran.orderid==Order.orderid,ArTran.transactioncodeid==ArTran.SALE,ArTran.contractid==self.contractid))\
            .all()

    @serialize_property_when("'_FOR_CLIENT_WRITE_' in self.__dict__")
    def financedorder(self):
        if self.currentorders:
            return self.currentorders[0]
        else:
            return None
```
financedorder will be None.

Either the original SAL transaction needs to be updated (which would also mean that reinstatement would have to put it back), or the report needs to be significantly altered.

This will not be a small thing...it means that whole data generator and report might have to be restructured from the ground up.

**Comment by kbower** [comment]
_**rneville** trac comment on May 8, 2012, 9:42:30 AM:_

(Changeset [[2636]](../commit/71e2f203eb8e71f788ea22eb9348ef07bf2d997a))
 * fixes #1636 (print contract button breaks on refinanced contract)

**Comment by kbower** [comment]
_**rneville** trac status change on May 8, 2012, 9:42:30 AM:_ **closed**
